oak_functions_service/src/instance.rs

```diff
@@ -76,13 +76,25 @@ impl<H: Handler> OakFunctionsInstance<H> {
                     micro_rpc::StatusCode::InvalidArgument,
                     "no chunk in extend request",
                 ),
-            )?));
+            )?))
+            .map_err(|msg| {
+                micro_rpc::Status::new_with_message(micro_rpc::StatusCode::InvalidArgument, msg)
+            })?;
         Ok(ExtendNextLookupDataResponse {})
     }
 
-    pub fn extend_lookup_data_chunk(&self, chunk: LookupDataChunk) {
+    pub fn extend_lookup_data_chunk(
+        &self,
+        chunk: LookupDataChunk,
+    ) -> Result<(), micro_rpc::Status> {
         self.lookup_data_manager
             .extend_next_lookup_data(to_data(chunk))
+            .map_err(|err| {
+                micro_rpc::Status::new_with_message(
+                    micro_rpc::StatusCode::InvalidArgument,
+                    format!("Was there a duplicate key?: {:?}", err),
```

Comment:
A more boring error message would unfortunately be more suitable here.

Reply:
Simplified.